build/hellobuild.md

```diff
@@ -0,0 +1,140 @@
+<br/>Docker builds images by reading the instructions from a Dockerfile.
+
+Docker images consist of **read-only layers**, each resulting from an instruction in the Dockerfile. Layers are stacked sequentially and each one is a delta representing the changes applied to the previous layer.
+
+## Dockerfile basics
+
+A Dockerfile is a text file containing all necessary instructions needed to assemble and deploy a container image with Docker.
+
+Here are the most common types of instructions:
+ 
+* [**FROM \<image\>**](../engine/reference/builder/#from) - defines a base for your image.
+* [**RUN \<command\>**](../engine/reference/builder/#run) - executes any commands in a new layer on top of the current image and commits the result. <br/>RUN also has a shell form for running commands.
+* [**WORKDIR \<directory\>**](../engine/reference/builder/#workdir) - sets the working directory for any RUN, CMD, ENTRYPOINT, COPY and ADD instructions that follow it in the Dockerfile.
+* [**COPY \<src\> \<dest\>**](../engine/reference/builder/#copy) - copies new files or directories from \<src\> and adds them to the filesystem of the container at the path \<dest\>.
+* [**CMD \<command\>**](../engine/reference/builder/#cmd) - lets you define the default program that is run once you start the container based on this image. Each Dockerfile only has one CMD, and only the last CMD instance is respected when multiple exist.
+<br/>Dockerfiles are crucial inputs for image builds and can facilitate automated, multi-layer image builds based on your unique configurations. Dockerfiles can start simple and grow with your needs and support images that require complex instructions.
+  
+For all the possible instructions, see the [Dockerfile reference](../engine/reference/builder/).
+
+## Example
+Here’s a simple Dockerfile example to get you started with building images. We’ll take a simple "Hello World" Python Flask application, and bundle it into a docker image that we can test locally or deploy anywhere!
+
+**Sample A**
+<br/>Let’s say we have the following in a `hello.py` file in our local directory:
+
+{% highlight python lineos %}
+from flask import Flask
+app = Flask(__name__)
+
+@app.route("/")
+def hello():
+    return "Hello World!"
+{% endhighlight %}
+
+Don’t worry about understanding the full example if you’re not familiar with Python - it’s just a simple web server that will contain a single page that says “Hello World”.
+
+> **Note** 
+>
+> If you test the example, make sure to copy over the indentation as well!
+
+**Sample B**
+<br/>Here’s a Dockerfile that Docker Build can use to create an image for our application:
+
+{% highlight dockerfile lineos %}
+# syntax=docker/dockerfile:1
+FROM ubuntu:22.04
+
+# install app dependencies
+RUN apt-get update && apt-get install -y python3 python3-pip
+RUN pip install flask
```

Comment:
```suggestion
RUN pip install flask==2.1.*
```

We should probably pin the flask version here so that these instructions should keep working even if flask makes backwards-incompatible changes. Additionally, using our own best-practices is a good idea :smile: